packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -51,10 +54,36 @@ class TransferEngine implements ITransferEngine {
     }
 
     throw new Error(
-      `Strapi versions doesn't match (${strategy} check): ${sourceVersion} does not match with ${destinationVersion} `
+      `Strapi versions doesn't match (${strategy} check): ${sourceVersion} does not match with ${destinationVersion}`
     );
   }
 
+  #assertSchemasMatching(sourceSchemas: any, destinationSchemas: any) {
+    const chosenStrategy = this.options.schemasMatching || 'strict';
+    const keys = uniq(Object.keys(sourceSchemas).concat(Object.keys(destinationSchemas)));
+    const diffs: { [key: string]: Diff[] } = {};
+
+    keys.forEach((key) => {
+      const sourceSchema = sourceSchemas[key];
+      const destinationSchema = destinationSchemas[key];
+      const schemaDiffs = calculateSchemaDiffs(sourceSchema, destinationSchema);
+
+      const isValid = schemaDiffs[chosenStrategy]();
+
+      if (!isValid) {
+        diffs[key] = schemaDiffs.diffs;
+      }
+    });
+
+    if (!isEmpty(diffs)) {
+      throw new Error(
+        `Import process failed because the project doesn't have a matching data structure 
```

Comment:
What does "the project" refers to here?

Reply:
It's like this in the acceptance criteria:

> when the schema doesn't correspond then the import process is stopped and a message in shown in the terminal `Import process failed because the project doesn't have a matching data structure` with the details of the differings with the destination schema